packages/client/src/query/WhereClause.ts

```diff
@@ -113,12 +128,18 @@ export type GeoFilter_Intersects = {
 
 export type GeoFilter = GeoFilter_Within | GeoFilter_Intersects;
 
-type FilterFor<PD extends ObjectTypePropertyDefinition> = PD["type"] extends
-  "string" ? StringFilter
-  : PD["type"] extends "geopoint" | "geoshape" ? GeoFilter
-  : PD["type"] extends "datetime" | "timestamp" ? DatetimeFilter
-  : PD["type"] extends "boolean" ? BooleanFilter
-  : NumberFilter; // FIXME we need to represent all types
+type FilterFor<PD extends ObjectTypePropertyDefinition> =
+  PD["multiplicity"] extends true
```

Comment:
I thought it was clearer to do the multiplicity check right here and split out into the different array filters, but could also do that check in the individual `StringFilter`, `GeoFilter`, `DateTimeFilter` etc. types instead

Reply:
I think you need `()` around the array cases, the ternary syntax gets weird. Even if its correct as is it would be clearer with parens.

Reply:
Yea good call, just added